### PR TITLE
fixed #10719: eigenvals of empty matrix raises IndexError

### LINKED ISSUE #10719: eigenvals of empty matrix raises IndexError
```python
>>> Matrix([[]]).eigenvals()
Traceback (most recent call last):
[...]
IndexError: Index out of range: a[0]
```
This appears to be a bug, spotted by @aravindkanna in https://github.com/sympy/sympy/issues/10701#issuecomment-191355482. I'd expect `eigenvals()` of an empty matrix to return an empty collection instead of raising an indexing exception.

### PR DESCRIPTION
**Pull request by rd13123013**
Now,

```
>>> Matrix([]).eigenvals()
{}
```

fixed #10719

### CODE REVIEW

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -3033,6 +3033,9 @@ def eigenvals(self, **flags):
         # unless the nsimplify flag indicates that this has already
         # been done, e.g. in eigenvects
         mat = self
+
+        if( not mat ):
```
Comment (leosartaj) [inline sympy/matrices/matrices.py]: No brackets are required here. ``if not mat`` should do.

File: sympy/matrices/tests/test_matrices.py [hunk 2]
```diff
@@ -927,6 +927,9 @@ def NS(e, n):
     sevals = list(sorted(evals.keys()))
     assert all(abs(nevals[i] - sevals[i]) < 1e-9 for i in range(len(nevals)))
 
+    # issue _
```
Comment (leosartaj) [inline sympy/matrices/tests/test_matrices.py]: `# issue _`  -> `# issue #10719`, maybe?

File: sympy/matrices/tests/test_matrices.py [hunk 3]
```diff
@@ -927,6 +927,9 @@ def NS(e, n):
...
+    assert Matrix([]).eigenvals() == {}
```
Comment (leosartaj) [inline sympy/matrices/tests/test_matrices.py]: ``assert Matrix([]).eigenvects() == []`` also works.

**Comment by leosartaj** [comment]
Other than my few comments. This looks good. + 1

**Comment by rd13123013** [comment]
Ping @leosartaj

**Comment by leosartaj** [comment]
LGTM. Merging.

**Comment by leosartaj** [comment]
Thanks @rd13123013 !

**Pull request merged**